src/api/config/genesis-params.md

```diff
@@ -0,0 +1,50 @@
+# Genesis Parameters
+
+Explain the purpose of this block. Maybe explain both keys in a single
+section?
+
+## `genesis.public_key`
+
+- **Type:** [Multi_hash](glossary#type-multi-hash)
+- **Required**
+
+The public key of the genesis account, should be supplied to all peers.
+
+```toml
+[genesis]
+public_key = "ed0120FAFCB2B27444221717F6FCBF900D5BE95273B1B0904B08C736B32A19F16AC1F9"
+```
+
+
+## `genesis.private_key`
+
+- **Type:** Table, [Private Key](glossary#type-private-key)
+- **Required** if the configured peer submits the genesis block, **optional** otherwise
+
+The private key of the genesis account, only needed for the peer that
+submits the genesis block.
+
+
+```toml
+[genesis]
+private_key = { digest = "ed25519", payload = "82886B5A2BB3785F3CA8F8A78F60EA9DB62F939937B1CFA8407316EF07909A8D236808A6D4C12C91CA19E54686C2B8F5F3A786278E3824B4571EF234DEC8683B" }
+```
+
+
+::: info
+
+This parameter is required if the peer being configured submits the
+genesis, i.e. if it is run with the [`--submit-genesis`](../cli#submit-genesis)
+ CLI argument.
+
+:::
+
+::: warning
+
+
+
+The warning will be printed if the
```

Comment:
What exactly does the warning say? What happens after the warning is printed?

Reply:
Let's come up with a line!

I want to remind that this configuration reference describes the configuration that doesn't exist yet. We are at the design stage, and we need to find the best ways to present configuration system to the users. It means that we might change the naming/structure/phrasing of anything if we decide it will be more convenient to users.

As for this topic, actually Iroha should fail to start with `--submit-genesis` arg, but without `genesis.private_key` config parameter. However, if there is no `--submit-genesis` arg, but `genesis.private_key` is provided, what Iroha should do? Just print a warning and continue execution, or fail?

@Mingela, your input might be valuable.

Reply:
Since the private key is needed only when a flag is specified it feels like the values should be combined somehow, i.e. `--submit-genesis --private-key <key>`. I might have encountered flag dependencies where one couldn't be used without other but cannot find an example. So following my suggestion just `--private-key <key>` must prompt an error and usage manual (like `usage: iroha [--submit-genesis --private-key <key>] .. etc`).
Another idea came to my mind, what about removing `--submit-genesis` at all and treat presence of `--private-key <key>` implicitly as it's intended to submit the genesis block?

Reply:
`private-key` was used for simplicity there, we should define a simpler name for `genesis.private_key` in that case, maybe `genesis-key`?

Reply:
There are two parameters:

- `genesis.public_key`: used by all peers
- `genesis.private_key`: used only by the peer which submits the genesis

Not sure how `genesis-key` (or `genesis.key`) might be used in this context.

The downside of `iroha --submit-genesis --private-key <key>`: it is not good to provide sensitive information through CLI. Although it might be workarounded with ENV: `--private-key $IROHA_PRIV_KEY`.

Anyway, the solution of omitting `--submit-genesis` seems fine. The main downside of it as I see is that it might not be that obvious which peer submits the genesis, and the user must know the details of how configuration works.

Still, if we want to choose the path of the highest reliability for the end user, I think the current approach is the best. You clearly identify your intention with `--submit-genesis` flag. Iroha will clearly fail if you forgot to provide `genesis.private_key` (or `GENESIS_PRIVATE_KEY` var (naming is up to discussion)). In this case, the question becomes: should Iroha **fail or warn** if the key is provided, but the intention was not given?

Reply:
Either is fine. Seems all options around the `--submit-genesis` have tradeoffs. I don't like the ability to have unused parameters specified, though a warning would be reasonable in that case. Let's hear opinions of others, @mversic @BAStos525

Reply:
We definitely should not provide `genesis.private.key` as shell parameter and it has to be hided somehow, e.g. env var. Currently, we have [this block](https://github.com/hyperledger/iroha/blob/df343f3390829121d193fa85fd10d540ccbe41d9/configs/peer/config.json#L50) in peer `config.json` file, and we pass these keys in plain text. Should it be hided somehow? Moreover, we don't separate this config file between peers, so it's the same for all peers. We manage genesis submitting only by `--submit-genesis` for `peer-0` selection by script. 
So, there are two ways:
  1. Should we pass genesis keys through `config.json`? Should it be hided and private key should be specified only for genesis peer-0 (different configs for peers then). And if it's necessary to use `--submit-genesis` flag for `peer-0` if keys are already specified in config?
  2. Remove `--submit-genesis` flag and this `genesis.private.key` field in config and apply genesis private key through ENV var or any kind of secret.

The second item sounds a kinda better since it may be easier to hide a sensitive data.
As for _should iroha if the keys are provided, but the intention was not given_, I think it depends on the selected approach how to provide the private key. If through config and flag, so it means to have a more deployment routine to separate peers configs e.t.c. to avoid iroha fail. So, here warning sounds more reasonable. In the second option we could try to fail it in order do not spread sensitive data across peers where and if it's not necessary.

Reply:
We had a discussion with the team and dove deeper into the topic of genesis. We agreed that the whole genesis approach as of now is somewhat weird and needs thoughtful revision. It will be done later (idk when, hopefully soon). As for configuration, I will preserve the current behavior without significant changes.

That is:

- `genesis.public_key` should be provided for all peers. Through config file **or** ENV var.
- `genesis.private_key` should be provided for the peer with `--submit-genesis` CLI flag. It might be set through config file **or** ENV var.
- if `genesis.private_key` is provided, but `--submit-genesis` is not, then Iroha prints a warning, but continues execution.
- if `--submig-genesis` is provided, but `genesis.private_key` is not, then Iroha fails with FATAL error.